kustomize/data_source_kustomization_overlay_test.go

```diff
@@ -1203,6 +1206,8 @@ data "kustomization_overlay" "test" {
 		version = "0.0.1"
 		namespace = "not-used"
 		skip_tests = true
+		kube_version = "1.29.0"
+		api_versions = ["monitoring.coreos.com/v1"]
```

Comment:
It seems to me this only adds the attributes to the tests but since they are never used, they aren't actually tested. Am I missing something?

Reply:
The test is located in the above attributes test:

https://github.com/FlxPeters/terraform-provider-kustomization/blob/d71d4c23d1a30618b4847772b88fed966f6eb904/kustomize/data_source_kustomization_overlay_test.go#L1188-L1191

We talked about the typo of test internaly a decided to not add a specific Helm test. The background is, that the provider code is only responsible for mapping the values on the [HelmChart](https://github.com/kubernetes-sigs/kustomize/blob/api/v0.17.1/api/types/helmchartargs.go#L37) struct. The implementation of the helm feature is located in kustomize code itself. So from my point of view it's correct to only thest the mapping.

Reply:
My understanding is, charts use these to conditionally include resources or not. So the test here must include a chart that has such a condition and verify that the conditional resource is included in the data source's manifest or not.

Reply:
The test - as is - makes sure the new attributes do not cause errors when passed to helm. The test case that you suggest would as well test that helm actually does the right thing. I would expect the helm project to have that functionality covered in their tests.

Reply:
In theory, you have a point. In practice, once I merge this, it becomes part of a code base I own. So if it breaks and a user opens an issue I can either ignore the user or sit down, invest my time, and debug it. To debug, I'd have to create a helm chart and use it via the provider to reproduce whatever the potential issue is.

Instead of creating future work for myself, I ask you to include a test in your contribution that has such a helm chart, and then I don't have to create a helm chart to reproduce myself or at least have a starting point.

For that very reason, there is already a helm chart in the tests and there are tests that use it to verify the manifests go from terraform to provider to kustomize/helm and all the way back to terraform without issues.

See: https://github.com/kbst/terraform-provider-kustomization/tree/master/kustomize/test_kustomizations/helm/initial/charts

Clearly, we disagree here. But, ultimately, this is my code base. So my opinion, in this very case, matters more than yours.

Time spent to add the test is better than time spent to argue with me why you shouldn't have to if you want to get this merged.

Reply:
Things you rarely hear on the internet: you were right! Thank you for your persistence, I found another mistake that would have slipped by me without the tests. There's now a Helm Chart specifically testing that feature.